File: January 2023/12 January 2023/.cpp
//{ Driver Code Starts
// Initial Template for C++

#include <bits/stdc++.h>
using namespace std;

// } Driver Code Ends  
//User function Template for C++

class Solution {
public:
    int minimizeSum(int N, vector<int> arr) {
      priority_queue<int,vector<int>,greater<int>>pq;
      for(int i=0;i<arr.size();i++){
          pq.push(arr[i]);
      }
      int ans=0;
      while(pq.size()>1){
           int sum = pq.top();
            pq.pop();
            sum += pq.top();
            pq.pop();
            pq.push(sum);
            ans += sum;
      }
       return ans;
    }
};
//{ Driver Code Starts.

int main() {
    int t;
    cin >> t;
    while (t--) {
        int N;
        cin >> N;
        vector<int> arr(N);
        for (int i = 0; i < N; i++) {
            cin >> arr[i];
        }
        Solution obj;
        cout << obj.minimizeSum(N,arr) << endl;
    }
}
// } Driver Code Ends

File: January 2023/20 January 2023/.py
#User function Template for python3

class Solution():
    def maxWeightCell(self, N, Edge):
        #your code goes here
  weight = [0]*N

        for i in range(N):

            if Edge[i] != -1:

                weight[Edge[i]] += i

        return weight.index(max(weight))
        #{
 # Driver Code Starts
#Initial Template for Python 3

if __name__ == "__main__":
    for _ in range(int(input())):
        N = int(input())
        Edge = [int(i) for i in input().split()]
        obj = Solution()
        ans=obj.maxWeightCell(N, Edge);
        print(ans)

# } Driver Code Ends
